addons/mail/models/ir_attachment.py

```diff
@@ -124,4 +125,13 @@ def _to_store(self, store: Store, /, *, fields=None, extra_fields=None):
                     if attachment.res_model != "mail.compose.message" and attachment.res_id
                     else False
                 )
+            if "message_id" in fields:
+                data["message_id"] = (
+                    Store.one(
+                        self.env['mail.message'].search([('attachment_ids', 'in', attachment.id)], limit=1),
+                        only_id=True,
+                    )
+                    if attachment.res_model != "mail.compose.message" and attachment.res_id
+                    else False
+                )
```

Comment:
- Why use `message_id` rather than existing `message` field in JS? This means the same so no need to have another field.
- I'm not sure this search is performant, especially on non-indexed column on very big table `mail.message`